gui-js/apps/minsky-electron/src/app/managers/CommandsManager.ts

```diff
@@ -1186,14 +1186,60 @@ export class CommandsManager {
       }
     });
   }
-  
-  static upgrade() {
-    let window=WindowManager.createWindow({
+
+  static startCSVDownload(payload: DownloadCSVPayload) {
+    const window=this.createDownloadWindow();
+
+    const splitUrl = payload.url.split('.');
```

Comment:
I think it would be more reliable to use the url package (https://nodejs.org/api/url.html) to grab the path name, then use the path package (https://nodejs.org/api/path.html) to extract the extension.

But even so, not all URLs even name the file, so it might be better to use the 'will-download' handler to grab the file name that the web service names it, and then pass the results back to the caller via a promise.